test/php/Nominatim/ParameterParserTest.php

```
@@ -0,0 +1,216 @@
+<?php
+
+namespace Nominatim;
+
+use Exception;
+
+require_once('../../lib/ParameterParser.php');
+
+
+function userError($sError)
+{
+    throw new Exception($sError);
+}
+
+class ParameterParserTest extends \PHPUnit_Framework_TestCase
+{
+
+
+    protected function setUp()
+    {
+    }
+
+
+    public function testGetBool()
+    {
+        $oParams = new ParameterParser([
+                                        'bool1' => '1',
+                                        'bool2' => '0',
+                                        'bool3' => 'true',
+                                        'bool4' => 'false',
+                                        'bool5' => ''
+                                       ]);
+
+        $this->assertSame(false, $oParams->getBool('non-exists'));
+        $this->assertSame(true, $oParams->getBool('non-exists', true));
+        $this->assertSame(true, $oParams->getBool('bool1'));
+        $this->assertSame(false, $oParams->getBool('bool2'));
+        $this->assertSame(true, $oParams->getBool('bool3'));
+        $this->assertSame(true, $oParams->getBool('bool4'));
+        $this->assertSame(false, $oParams->getBool('bool5'));
+    }
+
+
+    public function testGetInt()
+    {
+        $oParams = new ParameterParser([
+                                        'int1' => '5',
+                                        'int2' => '-1',
+                                        'int3' => '',
+                                        'int4' => 0
+                                       ]);
+
+        $this->assertSame(false, $oParams->getInt('non-exists'));
+        $this->assertSame(999, $oParams->getInt('non-exists', 999));
+        $this->assertSame(5, $oParams->getInt('int1'));
+
+        $this->assertSame(-1, $oParams->getInt('int2'));
+        $this->assertSame(false, $oParams->getInt('int3')); // FIXME: should be 0 instead?
```

This should throw for being an invalid value.